### DISCUSSION TITLE
Newb question

### DISCUSSION THREAD

**Discussion by mineralsman** [fairy-stockfish/Fairy-Stockfish#256] [General]
Hi, Ian. Thanks so much for your wonderful work! A friend and I are going to start learning to code games by making a chess GUI. Just to clarify, if we ever complete the project and release it, to bundle your engine with the GUI so the users have an engine to use (instead of having to add their own), we would have to make a directory accessible in file explorer with the source code + license text file in it? Would we also have to add one for stockfish or only for this engine? thanks again!

**Comment by ianfab** [comment]
Hi, thanks for your feedback.

The GPL only requires you to make the code accessible, not that you have to distribute it together with the binary. So if you just distribute it together with the license information and a link to the project on github this should be sufficient. Since Stockfish and Fairy-Stockfish use the same license and Fairy-Stockfish also retains the original copyright notices (in compliance with GPL) in the code from Stockfish, I think you do not need to directly refer to official Stockfish, but it does not hurt either.

When the GUI is distributed together with the engine, one could argue that the GUI then also has to be licensed under GPL (depends on whether it is considered "mere aggregation"), but I am not sure, since IANAL. If you anyway plan to do that, it's not an issue of course.
